src/shims/dlsym.rs

```diff
@@ -18,7 +19,8 @@ impl Dlsym {
     pub fn from_str<'tcx>(name: &[u8], target_os: &str) -> InterpResult<'tcx, Option<Dlsym>> {
         let name = &*String::from_utf8_lossy(name);
         Ok(match target_os {
-            "linux" | "macos" => unix::Dlsym::from_str(name, target_os)?.map(Dlsym::Posix),
+            target if target_os_is_unix(target) =>
+                unix::Dlsym::from_str(name, target_os)?.map(Dlsym::Posix),
```

Comment:
```suggestion
                unix::Dlsym::from_str(name, target)?.map(Dlsym::Posix),
```
Let's be consistent about the binder we use.